source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
I guess `spvOpcodeIsPointer` is meant to enforce the spec section 2.16.1.  In that case, it shouldn't be extended to `OpBitCast`, as the spec seems to forbid that.  BUT it should be limited to the Logical addressing model.  I presume OpenCL implementations will use a physical addressing model and won't be hampered by this.